### PR TITLE
Use pthread_cancel to terminate plugin threads

### PR DESCRIPTION
**Pull request by duke8253**

### CODE REVIEW

File: plugins/experimental/ssl_session_reuse/src/ssl_utils.h [hunk 1]
```diff
@@ -45,6 +47,35 @@ struct ssl_session_param {
...
+    while (!threads_queue.empty()) {
+      pthread_t th = threads_queue.front();
+      ::pthread_join(th, nullptr);
+      threads_queue.pop_front();
+    }
+  }
+
+private:
+  std::deque<pthread_t> threads_queue;
+  mutable std::mutex threads_mutex;
```
Comment (ywkaras) [inline plugins/experimental/ssl_session_reuse/src/ssl_utils.h]: Mutable member data is only useful in a class that will have const instances.  This class has no public interface that would be accessible for a const instance.  When would there be a const instance of this object?

Comment (duke8253) [reply plugins/experimental/ssl_session_reuse/src/ssl_utils.h]: I think you're right, removing mutable.

File: plugins/experimental/ssl_session_reuse/src/ssl_key_utils.cc [hunk 2]
```diff
@@ -368,6 +372,10 @@ STEK_update(const std::string &encrypted_stek)
 static void *
 STEK_Update_Checker_Thread(void *arg)
 {
+  plugin_threads.store(::pthread_self());
+  ::pthread_setcancelstate(PTHREAD_CANCEL_ENABLE, nullptr);
+  ::pthread_setcanceltype(PTHREAD_CANCEL_ASYNCHRONOUS, nullptr);
```
Comment (dragon512) [inline plugins/experimental/ssl_session_reuse/src/ssl_key_utils.cc]: how is this a good idea?

This leads to resource leaks as there is no way for the thread to respond to the "kill" of the thread. Should we not provide a message to the thread for it to read and shutdown with?

Comment (dragon512) [reply plugins/experimental/ssl_session_reuse/src/ssl_key_utils.cc]: I talked to the duke.. I understand this is a temp fix to a larger issue to address

**Pull request merged**